### ISSUE TITLE
-

### ISSUE THREAD

**Issue by luojiyin1987** [golangci/golangci-lint#6630]
## Problem

13 linter wrappers in `pkg/golinters/` repeat the same concurrency boilerplate (8+ lines each):

```go
var mu sync.Mutex
var resIssues []*goanalysis.Issue
// ...
mu.Lock()
resIssues = append(resIssues, issues...)
mu.Unlock()
// ...
WithIssuesReporter(func(*linter.Context) []*goanalysis.Issue { return resIssues })
```

This pattern appears in `nolintlint`, `unconvert`, `gocyclo`, `dupl`, `gocognit`, `gochecksumtype`, `goconst`, `promlinter`, `errcheck`, `gomodguard`, `gomodguard_v2`, `gosec`, `revive`, and `unused`.

## Solution

Introduce `ThreadSafeLinterBuilder` in `pkg/goanalysis/linter.go`:

```go
b := goanalysis.NewThreadSafeLinterBuilder()
b.Add(issues...)           // thread-safe append
b.Reporter()               // returns func(*linter.Context) []*Issue
```

## Impact

- 15 files changed
- +73 lines added, -210 lines removed
- Net reduction of 137 lines of boilerplate
- No behavioral change
- Full build passes

Makes adding new linter wrappers simpler and eliminates the risk of forgetting mutex locking.
